v0/destinations/personalize/transform.js

```diff
@@ -22,79 +16,64 @@ async function process(event) {
-              property[keyField[k]] = message[mappedFields];
-            } else {
-              throw new Error(`Mapped Field ${mappedFields} not found`);
-            }
-            break;
-          default:
-            traits = getFieldValueFromMessage(message, "traits");
-            keyField[k] = keyField[k]
-              .toLowerCase()
-              .replace(/_([a-z])/g, function (g) {
-                return g[1].toUpperCase();
-              });
-            if (traits[mappedFields]) {
-              property[keyField[k]] = traits[mappedFields];
-            } else if (
-              // if traits present in both roots and context
-              message.traits &&
-              message.context &&
-              message.context.traits
-            ) {
-              // retrieve traits inside context
-              context = message.context;
-              if (context.traits[mappedFields]) {
-                property[keyField[k]] = context.traits[mappedFields];
-              } else if (typeof event.message[mappedFields] !== "undefined") {
-                // if field is not present both in messsage.traits and message.context.traits we have search in the root
-                property[keyField[k]] = event.message[mappedFields];
-              }
-            } else if (typeof event.message[mappedFields] !== "undefined") {
-              // if no trait is present and we have search in the root
-              property[keyField[k]] = event.message[mappedFields];
-            } else {
-              throw new Error(`Mapped Field ${mappedFields} not found`);
-            }
-            break;
+
+        keyField[k] = keyField[k]
+          .toLowerCase()
+          .replace(/_([a-z])/g, function (g) {
+            return g[1].toUpperCase();
+          });
+        if (message.properties && message.properties[mappedFields]) {
+          property[keyField[k]] = message.properties[mappedFields];
+        } else if (typeof message[mappedFields] !== "undefined") {
+          property[keyField[k]] = message[mappedFields];
+        } else {
+          throw new Error(`Mapped Field ${mappedFields} not found`);
         }
       }
     }
+    const eventList = [];
+    const eventObj = {};
 
-    payload = {
-      eventList: [
-        {
-          eventId: event.message.messageId,
-          eventType: event.destination.Config.eventName,
-          properties: property,
-          sentAt: event.message.sentAt
-        }
-      ],
-      sessionId: event.message.originalTimestamp,
-      trackingId: event.destination.Config.trackingId,
-      userId: event.message.userId
-    };
+    if (property.eventValue) {
+      eventObj.eventValue = property.eventValue;
+      delete property.eventValue;
+    }
+    if (property.impression) {
+      eventObj.impression = property.impression;
+      delete property.impression;
+    }
+    if (property.itemId) {
+      eventObj.itemId = property.itemId;
```

Comment:
item_id is a required field in the schema. If it is not present set it to messageId